service/controller/aws/cluster.go

```diff
@@ -3,23 +3,29 @@ package aws
 import (
 	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
 	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/apprclient"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/giantswarm/operatorkit/client/k8scrdclient"
 	"github.com/giantswarm/operatorkit/controller"
 	"github.com/giantswarm/operatorkit/informer"
+	"github.com/spf13/afero"
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/giantswarm/cluster-operator/pkg/cluster"
+	"github.com/giantswarm/cluster-operator/pkg/v1/guestcluster"
```

Comment:
IMO this should be `v2`. Same problem below in a lot of places. It would be nice to prevent versioning on this level. So far the resource sets did the versioning completely and we only had to bother with one dedicated entity version. This would be here `aws/v1`, `aws/v2` etc.. So maybe we could just inject the logger and the cert searcher and create the `guestcluster` magic in the resource set. The versioning on this level will be way easier because the imports are all the same version and there is actually only one version per entity/package. The `goimports` integrations are able to sort it that way a lot better.